backend/connector_v2/models.py

```diff
@@ -111,17 +111,42 @@ def __str__(self) -> str:
             f" workflow: {self.workflow})"
         )
 
+    def get_connector_metadata_bytes(self):
+        """Convert connector_metadata to bytes if it is a memoryview.
+
+        Returns:
+            bytes: The connector_metadata as bytes.
+        """
+        if isinstance(self.connector_metadata, memoryview):
+            return self.connector_metadata.tobytes()
+        return self.connector_metadata
+
     @property
     def metadata(self) -> Any:
+        """Decrypt and return the connector metadata as a dictionary.
+
+        This property handles the decryption of the connector_metadata,
+        converting it to bytes if necessary, and then loading the decrypted
+        JSON string into a dictionary.
+
+        Returns:
+            dict: The decrypted connector metadata.
+        """
+
+        connector_metadata_bytes = self.get_connector_metadata_bytes()
+
+        if connector_metadata_bytes is None:
+            return None
```

Comment:
@muhammad-ali-e when can this be None? Let's handle errors when the metadata is NULL within the model itself rather than doing it at the caller side

Reply:
I think, the current approach is safer because there may be scenarios where connector_metadata could be None, especially in existing cases that I am  not fully aware of it (We can check it later once the migrations are done). Handling it within the model ensures that we avoid potential errors at the caller side.